lib/core/utils/node-serializer.js

```diff
@@ -0,0 +1,133 @@
+import assert from './assert';
+import DqElement from './dq-element';
+
+let customSerializer = null;
+
+const nodeSerializer = {
+  /**
+   * @param {Object} newSerializer
+   * @property {Function} toSpec (Optional) Converts an Element or VirtualNode to a "spec", a form
+   * suitable for JSON.stringify to consume. Output must include all properties
+   * that DqElement.toJSON() would have. Will always be invoked from the
+   * input element's original page context.
+   * @property {Function} mergeSpecs (Optional) Merges two specs (produced by toSpec) which
```

Comment:
Should this also include a note about needing to pass through all properties that `DqElement.mergeSpec` would pass through  (`ancestry`, `selector`, etc.) just like the note for `toSpec`?

Reply:
I don't think that's crucial, but if you have a suggestion I'll put it in 😁